sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java

```diff
@@ -51,7 +51,7 @@
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         return Mono.defer(() -> {
             if (cachedToken.get() != null) {
-                return identityClient.authenticateWithUserRefreshToken(request, cachedToken.get())
+                return identityClient.authenticateWithMsalAccount(request, cachedToken.get().getAccount())
                     .onErrorResume(t -> Mono.empty());
```

Comment:
Would you want this to resume on an error and return an empty Mono? I'd expect the error to be propagated downstream and allow downstream subscribers to do with that error what they will. In  `globalSetupAsync` this is call is chained with `.then()`, so it would hide the error.

There are similar instances in other credentials below.

Reply:
Yeah it should be probably be handled downstream - it's out of the scope of this PR but I can see if it can be done with minimal footprint